google_rest_api_generator/src/method_builder.rs

```diff
@@ -247,22 +249,36 @@ fn path_method(
     }
 }
 
-fn reqwest_http_method(http_method: &str) -> syn::Path {
-    match http_method {
-        "GET" => parse_quote! {::reqwest::Method::GET},
-        "POST" => parse_quote! {::reqwest::Method::POST},
-        "PUT" => parse_quote! {::reqwest::Method::PUT},
-        "DELETE" => parse_quote! {::reqwest::Method::DELETE},
-        "HEAD" => parse_quote! {::reqwest::Method::HEAD},
-        "OPTIONS" => parse_quote! {::reqwest::Method::OPTIONS},
-        "CONNECT" => parse_quote! {::reqwest::Method::CONNECT},
-        "PATCH" => parse_quote! {::reqwest::Method::PATCH},
-        "TRACE" => parse_quote! {::reqwest::Method::TRACE},
+fn reqwest_http_method(http_method: &::reqwest::Method) -> syn::Path {
+    match *http_method {
+        ::reqwest::Method::GET => parse_quote! {::reqwest::Method::GET},
+        ::reqwest::Method::POST => parse_quote! {::reqwest::Method::POST},
+        ::reqwest::Method::PUT => parse_quote! {::reqwest::Method::PUT},
+        ::reqwest::Method::DELETE => parse_quote! {::reqwest::Method::DELETE},
+        ::reqwest::Method::HEAD => parse_quote! {::reqwest::Method::HEAD},
+        ::reqwest::Method::OPTIONS => parse_quote! {::reqwest::Method::OPTIONS},
+        ::reqwest::Method::CONNECT => parse_quote! {::reqwest::Method::CONNECT},
+        ::reqwest::Method::PATCH => parse_quote! {::reqwest::Method::PATCH},
+        ::reqwest::Method::TRACE => parse_quote! {::reqwest::Method::TRACE},
         _ => panic!("unknown http method: {}", http_method),
     }
 }
 
-fn request_method<'a>(http_method: &str, params: impl Iterator<Item = &'a Param>) -> TokenStream {
+fn method_auth_scope<'a>(http_method: &::reqwest::Method, scopes: &'a [String]) -> Option<&'a String> {
```

Comment:
Minor nit: Let's make this return Option<&str> instead. I tend to avoid &String in function signatures.

Reply:
Done